drivers/flash/flash_cadence_qspi_nor.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <kernel.h>
+#include <device.h>
+
+#define DT_DRV_COMPAT cdns_qspi_nor
+
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+
+#include "flash_cadence_qspi_nor_ll.h"
+
+struct flash_cad_priv {
+	DEVICE_MMIO_RAM;
+	struct cad_qspi_params params;
+};
+
+struct flash_cad_config {
+	DEVICE_MMIO_ROM;
+	uintptr_t cfg_data_reg;
+	uint32_t cfg_data_size;
+};
+
+static const struct flash_parameters flash_cad_parameters = {
+	.write_block_size = QSPI_BYTES_PER_DEV,
+	.erase_value = 0xff,
+};
+
+static int flash_cad_read(const struct device *dev, off_t offset,
+				void *data, size_t len)
+{
+	int rc;
+
+	if (dev == NULL) {
+		printk("Error: No device detected");
```

Comment:
This should be using a LOG and not printk throughout this whole file (and others)

Reply:
Addressed. Thanks.

Reply:
@XenuIsWatching i know why i'm not using the LOG at the first place, it was unrelated error when running twister test, that forces me to use printk. can you advice how should i resolve the error in the twister?